src/codec2.c

```diff
@@ -114,7 +114,7 @@ struct CODEC2 * codec2_create(int mode)
     struct CODEC2 *c2;
     int            i,l;
 
-#ifndef CORTEX_M4
+#ifdef CORTEX_M4
     if (( CODEC2_MODE_ACTIVE(CODEC2_MODE_450, mode)) || 
```

Comment:
@der-stefan - this appears to disable 450 bit/s mode for PC builds?

Reply:
No, the exact opposite is true. Only if CORTEX_M4 is defined, the 450 mode should be disabled. It wasn't me who added this switch - but there must be a reason why it is there. Perhaps because all the lookup table should not be compiled into the ARM program by default.

Reply:
Ahh yes I see, the function is currently returning NULL on PC builds, thereby failing to allow 450 mode to start.  Perhaps we don't need the entire if/then clause, as 450 mode won't be compiled in at all for CORTEX_M4. Yes - many modes are not compiled for CORTEX_M4 to save program memory on our target platforms.

@db4ple do you have any comments?

Reply:
@der-stefan - @db4ple (Danilo) is the master of the macros, and did some great work on the CORTEX_M4 support, he also works on the UHSDR project.

Reply:
Hi all, the current code is clearly a mistake. The proposed change makes sense. The current code doesn't as it indeed prevents use of the 450 modes on anything but a CORTEX_M4.

We will have to look at little more into it to decide if we could leave out the #ifdef/#ifndef CORTEX_M4 at all in codec2.c. 

There is a general problem with the codec2_create as it is. It may create a not fully initialized codec2 structure (e.g. if using the CODEC2_700) which is difficult to detect in the applications code. 
We should block all calls with unsupported (read not included in this build) CODEC2_xxxx modes right at the beginning of codec2_create like we do in the freedv_open. We can then get rid of all CORTEX_M4 exclusions in codec2_create.

In a nutshell: We should merge this.

Reply:
@db4ple thanks, I'll run the stm32/unittests then merge.  Stefan - these tests use stm32f4 hardware so we run them as a manual step outside of Github/Travis.